packages/plugin-drizzle/src/index.ts

```diff
@@ -315,6 +317,15 @@ export function drizzleStorage<
               TSchema
             >;
 
+            if (prevFinality === "pending") {
+              // invalidate if previous block's finality was "pending"
+              await invalidate(tx, cursor, idColumn, indexerId);
+
+              if (enablePersistence) {
+                await invalidateState({ tx, cursor, indexerId });
+              }
```

Comment:
This one is not needed since persisted state is not updated when data is pending.